xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -215,6 +222,12 @@ public void updateLocalityStore(Map<XdsLocality, LocalityInfo> localityInfoMap)
         ChildHelper childHelper;
         if (oldLocalities.contains(newLocality)) {
           LocalityLbInfo oldLocalityLbInfo = localityMap.get(newLocality);
+
+          if (oldLocalityLbInfo.isDeactivated()) {
+            oldLocalityLbInfo.reactivate();
+            loadStatsStore.addLocality(newLocality);
```

Comment:
We can also delete the check in `loadStatsStore.addLocality()` to relax the calling requirements of `addLocality()`, and then we can move this line (and the one in the else-branch) outside of the if-block, and that would be simpler.

Reply:
After offline discussion with @voidzcy we prefer to only check EDS response changes.